editor/src/core/shared/math-utils.ts

```diff
@@ -1020,3 +1020,7 @@ export function segmentIntersection(
   }
   return null
 }
+
+export function mod(n: number, m: number) {
+  return ((n % m) + m) % m
```

Comment:
what is this? why isn't % enough? is this some JavaScript madness here?

Reply:
unfortunately the % is the remainder operator in JS which works differently from modulo for negative values. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Remainder#description